### DISCUSSION TITLE
When a method is executed, the subsequent processing is not executed.

### DISCUSSION THREAD

**Discussion by QuestDragon** [scripthookvdotnet/scripthookvdotnet#1399] [Q&A] [answered]
I'm a beginner at creating scripts, so I may not understand the basics, but I'd like to ask for your help, so I'm posting this.

I tried to make the code look better by grouping the same processing into a method, but when I call a method inside an event handler method such as OnTick or OnKeyDown, **it doesn't execute.**

What should I do?

_(I'm Japanese and I'm using Google Translate for this text. Please forgive me if there are any parts that are clumsy.)_

![2024-02-05](https://github.com/scripthookvdotnet/scripthookvdotnet/assets/126067454/890b101d-0cec-4890-9438-556f18b6e781)
![20240205010730_1](https://github.com/scripthookvdotnet/scripthookvdotnet/assets/126067454/4ea278a0-5800-40ed-8546-69ec97cf662d)

**Comment by kagikn** [comment, marked as answer]
You can't use `await` in methods where native functions are called at the moment. Await works like how it works in console apps currently. SHVDN doesn't set up a custom task scheduler where `Task`s run on a specific thread with a custom synchronization context, and that prevents from resuming jobs on a specific thread. If you have to use async `Task`s, you're going to have to start them manually with methods such as `Task.Run` and manually check the `Task.IsCompleted` every tick you want to test. I think you'll end up using some state variables to manage "async" tasks in SHVDN scripts, too.

**Comment by QuestDragon** [comment]
I see...I understand! I will try it. I will ask you more questions if you have any more questions.

**Reply by QuestDragon** [reply]
It's done! thank you very much!
